### PR TITLE
Quote whitespace-containing arguments in Task.description

### PR DESCRIPTION
**Pull request by ikesyo**
Fixes https://github.com/Carthage/Carthage/issues/1318.

### CODE REVIEW

File: ReactiveTask/Task.swift [hunk 1]
```diff
@@ -48,6 +48,13 @@ public struct Task {
 
 extension Task: CustomStringConvertible {
 	public var description: String {
+		let arguments = self.arguments.map { argument -> String in
```
Comment (NachoSoto) [inline ReactiveTask/Task.swift]: Maybe extract this into a `escape` function in a `String` extension?

File: ReactiveTask/Task.swift [hunk 2]
```diff
@@ -48,6 +48,13 @@ public struct Task {
...
+			if argument.containsString(" ") {
```
Comment (mdiep) [inline ReactiveTask/Task.swift]: It would be better to use `NSCharacterSet` to test for whitespace—that will catch other whitespace characters like `\t`.

**Comment by ikesyo** [comment]
Updated ✨ How does it look like now?

File: ReactiveTask/Task.swift [hunk 3]
```diff
@@ -46,16 +46,22 @@ public struct Task {
...
 
-extension Task: CustomStringConvertible {
-	public var description: String {
-		let arguments = self.arguments.map { argument -> String in
-			if argument.containsString(" ") {
-				return "\"\(argument)\""
+private extension CollectionType where Generator.Element == String {
+	func escaped() -> [String] {
+		let whitespaceCharacterSet = NSCharacterSet.whitespaceCharacterSet()
+		return map { string -> String in
```
Comment (NachoSoto) [inline ReactiveTask/Task.swift]: The return type here is redundant.

File: ReactiveTask/Task.swift [hunk 4]
```diff
@@ -46,9 +46,22 @@ public struct Task {
 	}
 }
 
+private extension CollectionType where Generator.Element == String {
+	func escaped() -> [String] {
+		let whitespaceCharacterSet = NSCharacterSet.whitespaceCharacterSet()
+		return map { string -> String in
+			if string.rangeOfCharacterFromSet(whitespaceCharacterSet) != nil {
```
Comment (NachoSoto) [inline ReactiveTask/Task.swift]: I would extract this to a method in terms of `String`. I think it doesn't make sense to have the canonical implementation for this on a `CollectionType` extension.

File: ReactiveTask/Task.swift [hunk 5]
```diff
@@ -46,9 +46,22 @@ public struct Task {
 	}
 }
 
+private extension CollectionType where Generator.Element == String {
```
Comment (NachoSoto) [inline ReactiveTask/Task.swift]: This could be `SequenceType` instead.

File: ReactiveTask/Task.swift [hunk 6]
```diff
@@ -46,9 +46,22 @@ public struct Task {
...
+	func escaped() -> [String] {
```
Comment (NachoSoto) [inline ReactiveTask/Task.swift]: I think this would make more sense as a derived property. What do you think?

File: ReactiveTask/Task.swift [hunk 7]
```diff
@@ -46,9 +46,20 @@ public struct Task {
 	}
 }
 
+private extension String {
+	var escaped: String {
+		let whitespaceCharacterSet = NSCharacterSet.whitespaceCharacterSet()
```
Comment (ikesyo) [inline ReactiveTask/Task.swift]: Calling this for each call would be okay?

**Comment by mdiep** [comment]
🚢

**Pull request merged**